llama_index/tools/types.py

```diff
@@ -112,3 +112,56 @@ def to_langchain_structured_tool(
             func=self.__call__,
             **langchain_tool_kwargs,
         )
+
+
+class AsyncBaseTool(BaseTool):
```

Comment:
So just to confirm, the intention for future tools would be to encourage implementing this class?

Reply:
Yup! This would be the new parent class for all new tools moving forward